pkg/apis/workspaces/v1alpha1/commands_conversion.go

```diff
@@ -40,8 +40,36 @@ func convertCommandTo_v1alpha2(src *Command, dest *v1alpha2.Command) error {
 	return nil
 }
 
+// getGroup returns the group the command belongs to
+func getGroup(dc v1alpha2.Command) *v1alpha2.CommandGroup {
+	switch {
+	case dc.Composite != nil:
+		return dc.Composite.Group
+	case dc.Exec != nil:
+		return dc.Exec.Group
+	case dc.Apply != nil:
+		return dc.Apply.Group
+	case dc.Custom != nil:
+		return dc.Custom.Group
+
+	default:
+		return nil
+	}
+}
+
 func convertCommandFrom_v1alpha2(src *v1alpha2.Command, dest *Command) error {
+	if src == nil {
+		return nil
+	}
+
 	id := src.Key()
+
+	srcCmdGroup := getGroup(*src)
+	if srcCmdGroup != nil && srcCmdGroup.Kind == v1alpha2.DeployCommandGroupKind {
+		// skip converting deploy kind commands as deploy kind commands are not supported in v1alpha1
+		return nil
+	}
+
```

Comment:
I'm wondering how best to approach this. The way this is done breaks round-tripping through `v1alpha1` if a DevWorkspace uses deploy commands (in Kubernetes, all API versions should be safely convertible to any other version without loss of data).

We could add `deploy` commands to v1alpha1 (I think that's still compatible) and let it be converted freely, but this does feel like somewhat of an unnecessary burden for a situation that should not be encountered (saying that, though, means it definitely *will* be encountered :upside_down_face:)

Do we have a policy in place for things like this? 

Reference: https://book.kubebuilder.io/multiversion-tutorial/api-changes.html

> In Kubernetes, **all versions must be safely round-tripable through each other**. This means that if we convert from version 1 to version 2, and then back to version 1, we must not lose information. Thus, any change we make to our API must be compatible with whatever we supported in v1, and also need to make sure anything we add in v2 is supported in v1. In some cases, this means we need to add new fields to v1, but in our case, we won’t have to, since we’re not adding new functionality.

Reply:
AFAIK, I dont think we do have a policy in place.

I just checked and it seems we also lose `variables` which is defined under `DevWorkspaceTemplateSpecContent`. I'll add up a topic so we can have a chat about this in our cabal meeting

Reply:
I see you already added the topic 😄